components/director/internal/domain/labeldef/service.go

```diff
@@ -60,3 +75,55 @@ func (s *service) List(ctx context.Context, tenant string) ([]model.LabelDefinit
 	}
 	return defs, nil
 }
+
+func (s *service) Update(ctx context.Context, def model.LabelDefinition) error {
+	if err := def.ValidateForUpdate(); err != nil {
+		return errors.Wrap(err, "while validating Label Definition")
+	}
+
+	ld, err := s.repo.GetByKey(ctx, def.Tenant, def.Key)
+	if err != nil {
+		return errors.Wrap(err, "while receiving Label Definition")
+	}
+
+	if ld == nil {
+		return errors.Errorf("definition with %s key doesn't exist", def.Key)
+	}
+
+	ld.Schema = def.Schema
+
+	if def.Schema == nil {
+		if err := s.repo.Update(ctx, *ld); err != nil {
```

Comment:
Please refactor the code that the repository Update method is called once.

Reply:
done